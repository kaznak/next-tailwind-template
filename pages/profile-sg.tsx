import useAuth from 'lib/useAuth'
import { useState } from 'react'
import { fetchJson, FetchError } from 'lib/fetchJson'

export default function SgProfile() {
  const { auth, mutateAuth } = useAuth({
    redirectTo: '/signin',
  })
  const [errorMessage, setErrorMessage] = useState('')
  const signOut = async () => {
    try {
      mutateAuth(
        await fetchJson('/api/auth', {
          method: 'DELETE',
          headers: { 'Content-Type': 'application/json' },
        })
      )
    } catch (error) {
      if (error instanceof FetchError) {
        setErrorMessage(error.data.message)
      } else {
        console.error('An unexpected error happened:', error)
      }
    }
  }

  return (
    <div className="container">
      <p>{JSON.stringify(auth)}</p>
      <button onClick={signOut}>Sign Out</button>
      {errorMessage && <p className="error">{errorMessage}</p>}
    </div>
  )
}
