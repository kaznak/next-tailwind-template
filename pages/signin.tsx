import { useState, FormEvent } from 'react'
import useAuth from 'lib/useAuth'
import { fetchJson, FetchError } from 'lib/fetchJson'

export const SignIn = () => {
  const { mutateAuth } = useAuth({
    redirectTo: '/profile-sg',
    redirectIfFound: true,
  })
  const [errorMessage, setErrorMessage] = useState('')

  const onSubmit = async (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    const body = {
      email: event.currentTarget.email.value,
    }
    try {
      mutateAuth(
        await fetchJson('/api/auth', {
          method: 'POST',
          headers: { 'Content-Type': 'application/json' },
          body: JSON.stringify(body),
        })
      )
    } catch (error) {
      if (error instanceof FetchError) {
        setErrorMessage(error.data.message)
      } else {
        console.error('An unexpected error happened:', error)
      }
    }
  }

  return (
    <div className="container">
      <form onSubmit={onSubmit}>
        <label>
          <span>Type your Email</span>
          <input type="text" name="email" required />
        </label>

        <button type="submit">Sign In</button>

        {errorMessage && <p className="error">{errorMessage}</p>}
      </form>
    </div>
  )
}

export default SignIn
