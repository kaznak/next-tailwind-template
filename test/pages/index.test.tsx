import React from 'react'
import { render, fireEvent } from 'test/testUtils'
import { Home } from 'pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('clicking button triggers alert', () => {
    const { getByText } = render(<Home />, {})
    window.alert = jest.fn()
    fireEvent.click(getByText('API Test Button'))
    expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  })
})
