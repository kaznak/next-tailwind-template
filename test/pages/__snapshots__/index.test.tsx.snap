// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center min-h-screen py-2"
  >
    <main
      class="flex flex-col items-center justify-center w-full flex-1 px-20 text-center"
    >
      <h1
        class="text-6xl font-bold"
      >
        Welcome to 
        <a
          class="text-blue-600"
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="my-3 text-2xl"
      >
        Get started by editing 
        <code
          class="p-3 font-mono text-lg bg-gray-100 rounded-md"
        >
          pages/index.js
        </code>
      </p>
      <button
        class="my-3 px-8 py-3 border border-transparent text-base font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 md:py-4 md:text-lg md:px-10"
      >
        Test Button
      </button>
      <div
        class="flex flex-wrap items-center justify-around max-w-4xl mt-6 sm:w-full"
      >
        <a
          class="p-6 mt-6 text-left border w-96 rounded-xl hover:text-blue-600 focus:text-blue-600"
          href="https://nextjs.org/docs"
        >
          <h3
            class="text-2xl font-bold"
          >
            Documentation →
          </h3>
          <p
            class="mt-4 text-xl"
          >
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="p-6 mt-6 text-left border w-96 rounded-xl hover:text-blue-600 focus:text-blue-600"
          href="https://nextjs.org/learn"
        >
          <h3
            class="text-2xl font-bold"
          >
            Learn →
          </h3>
          <p
            class="mt-4 text-xl"
          >
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="p-6 mt-6 text-left border w-96 rounded-xl hover:text-blue-600 focus:text-blue-600"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h3
            class="text-2xl font-bold"
          >
            Examples →
          </h3>
          <p
            class="mt-4 text-xl"
          >
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="p-6 mt-6 text-left border w-96 rounded-xl hover:text-blue-600 focus:text-blue-600"
          href="https://vercel.com/import?filter=next.js&utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h3
            class="text-2xl font-bold"
          >
            Deploy →
          </h3>
          <p
            class="mt-4 text-xl"
          >
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="flex items-center justify-center w-full h-24 border-t"
    >
      <a
        class="flex items-center justify-center"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <img
          alt="Vercel Logo"
          class="h-4 ml-2"
          src="/vercel.svg"
        />
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
